code/ship/ship.cpp

```diff
@@ -9072,46 +9127,54 @@ static void ship_auto_repair_frame(int shipnum, float frametime)
 	sip = &Ship_info[sp->ship_info_index];
 	objp = &Objects[sp->objnum];
 
-	if (sp->flags[Ship::Ship_Flags::Dying]) // do not repair if already dead 
+	// do not repair if already dead 
+	if (sp->flags[Ship::Ship_Flags::Dying])
 		return;
 
-	//Repair the hull...or maybe unrepair?
-	if(sip->hull_repair_rate != 0.0f)
+	// Repair the hull...or maybe unrepair?
+	if (!fl_equal(sip->hull_repair_rate, 0.0f))
 	{
-		objp->hull_strength += sp->ship_max_hull_strength * sip->hull_repair_rate * frametime;
+		float max_hull_repair_strength = sp->ship_max_hull_strength * sip->hull_repair_max;
+		objp->hull_strength += max_hull_repair_strength * sip->hull_repair_rate * frametime;
 
-		if (objp->hull_strength > sp->ship_max_hull_strength)
-			objp->hull_strength = sp->ship_max_hull_strength;
+		if (objp->hull_strength > max_hull_repair_strength)
+			objp->hull_strength = max_hull_repair_strength;
 		else if (objp->hull_strength < 0)
 			ship_hit_kill(objp, nullptr, nullptr, 0, true);
 	}
 
-	// only allow for the auto-repair of subsystems on small ships
-	//...NOT. Check if var has been changed from def -C
-	if ( !(sip->is_small_ship()) && sip->subsys_repair_rate == -2.0f)
+	// do not repair if it is a large ship with a default repair rate
+	if (!(sip->is_small_ship()) && sip->subsys_repair_rate == -2.0f)
 		return;
 	
-	if(sip->subsys_repair_rate == -2.0f)
-		real_repair_rate = SHIP_REPAIR_SUBSYSTEM_RATE;
```

Comment:
You said it was unused but it appears `SHIP_REPAIR_SUBSYSTEM_RATE` was used right here? Even if it's only used once, keeping magic numbers away from the code is still a good idea.